### PR TITLE
WIP: Trying to add count-value as an attribute of userCollection

### PR DESCRIPTION
**Pull request by the-good-boy**

### CODE REVIEW

**Comment by the-good-boy** [comment]
@MonkeyDo Is my implementation of the initialize function correct?

**Review by MonkeyDo — COMMENTED** [review]
I found what was missing: we needed to add a "group by" clause to the SQL query. That's done easily with knex with `query.groupBy(column)`.
I also put this functionality behind a `withItemCount` option flag so that it is opt-in.

To use it you just need to add it to the `fetchPage`  or `fetchAll` options: 
For example in `getOrderedPublicCollections`:
```
const allCollections = await new UserCollection()
		[…]
		.fetchPage({
			limit: size,
			offset: from,
			withItemCount: true, // <--- this guy !
			withRelated: ['owner']
		});
```

And you'll have an itemCount property available in each collection object!

File: src/models/userCollection.js [hunk 1]
```diff
@@ -26,6 +26,20 @@ export default function userCollection(bookshelf) {
...
 		idAttribute: 'id',
+		initialize() {
+			this.on('fetching fetching:collection', (model, columns, options) => {
+				if (options.withItemCount === true) {
+					options.query.select(bookshelf.knex.raw(
+						`(select count(*)
+						from bookbrainz.user_collection_item
+						where
+						collection_id = bookbrainz.user_collection.id
+						) as item_count`
```
Comment (MonkeyDo) [inline src/models/userCollection.js:37]: note it is `item_count`; the ORM will transform it into `itemCount` automatically.

Comment (the-good-boy) [reply src/models/userCollection.js:37]: Ah, nice! It works now. Thank you for helping me with this :)

**Review by MonkeyDo — COMMENTED** [review]
I'd like to see a test added for this functionality in this repository.

Once that is done, I'll merge the PR and publish a new version of the package that we can then update in package.json in the PR https://github.com/bookbrainz/bookbrainz-site/pull/571

**Comment by the-good-boy** [comment]
Alright! Right now, we dont have tests for userCollection, so I guess I'll try writing it from scratch and see how it goes.

**Comment by MonkeyDo** [comment]
> Alright! Right now, we dont have tests for userCollection, so I guess I'll try writing it from scratch and see how it goes.

I'm back from a week of vacations so don't hesitate if you have questions!

**Pull request merged**